crates/starknet-devnet-types/src/rpc/block.rs

```diff
@@ -84,6 +92,18 @@ pub struct BlockHeader {
     pub l1_da_mode: L1DataAvailabilityMode,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct PendingBlockHeader {
```

Comment:
Even tho we have BlockHeader with some redundant properties seems that we need to have a specific struct to handle responses right?

Maybe both should share a common struct like `#[serde(flatten)] header: Base BlockHeader `? What do you think guys?

(Same question applies to PendingStateUpdate)

Reply:
I tried to fix it in b6278bb38b743282479fc4a78eecfc9fa0d20ac0 but I reverted that later, apparently, there is a problem with serde flatten inside flatten